diff -Naurp vdr-plugin-satip-5e524a9/Makefile vdr-plugin-satip-5e524a9-patch/Makefile
--- vdr-plugin-satip-5e524a9/Makefile	2017-06-25 14:21:41.000000000 +0300
+++ vdr-plugin-satip-5e524a9-patch/Makefile	2017-07-03 20:24:35.000000000 +0300
@@ -53,7 +53,7 @@ SOFILE = libvdr-$(PLUGIN).so
 
 ### Libraries
 
-LIBS = $(shell curl-config --libs)
+LIBS = $(shell pkg-config --libs libcurl)
 
 ### Includes and Defines (add further entries here):
 
