diff -Naurp Alexelec-settings-30c0064/language/English/strings.po Alexelec-settings-30c0064-patch/language/English/strings.po
--- Alexelec-settings-30c0064/language/English/strings.po	2018-02-01 18:55:11.000000000 +0200
+++ Alexelec-settings-30c0064-patch/language/English/strings.po	2018-02-01 19:06:06.000000000 +0200
@@ -1118,6 +1118,44 @@ msgctxt "#2851"
 msgid "Select the shell loaded at startup."
 msgstr ""
 
+########## TVIP #########
+
+msgctxt "#32860"
+msgid "TVIP configuration"
+msgstr ""
+
+msgctxt "#32861"
+msgid "Enable playlist"
+msgstr ""
+
+msgctxt "#2861"
+msgid "Use the specified channel source (m3u-playlist)."
+msgstr ""
+
+msgctxt "#32862"
+msgid "M3U file:"
+msgstr ""
+
+msgctxt "#2862"
+msgid "Select *.m3u file."
+msgstr ""
+
+msgctxt "#32863"
+msgid "Update playlist"
+msgstr ""
+
+msgctxt "#2863"
+msgid "Update Torrent-TV playlist before TVIP start."
+msgstr ""
+
+msgctxt "#32864"
+msgid "First channel"
+msgstr ""
+
+msgctxt "#2864"
+msgid "Start TVIP from the first channel."
+msgstr ""
+
 ########## ACESTREAM #########
 
 msgctxt "#34000"
@@ -1353,7 +1391,7 @@ msgid "Use external IP-address in playli
 msgstr ""
 
 msgctxt "#34062"
-msgid "For"
+msgid "For:"
 msgstr ""
 
 msgctxt "#3462"
@@ -1361,13 +1399,37 @@ msgid "Playlist for AceStream and/or Nox
 msgstr ""
 
 msgctxt "#34063"
-msgid "Execute..."
+msgid "Execute ->"
 msgstr ""
 
 msgctxt "#3463"
 msgid "Create a playlist for Torrent-TV (/storage/ttv-m3u/)."
 msgstr ""
 
+msgctxt "#34064"
+msgid "Login"
+msgstr ""
+
+msgctxt "#3464"
+msgid "Login from site http://torrent-tv.ru"
+msgstr ""
+
+msgctxt "#34065"
+msgid "Password"
+msgstr ""
+
+msgctxt "#3465"
+msgid "Password from site http://torrent-tv.ru"
+msgstr ""
+
+msgctxt "#34066"
+msgid "Channel Source:"
+msgstr ""
+
+msgctxt "#3466"
+msgid "Channel source for AceStream: Torrent-TV, AceStream or both (All)."
+msgstr ""
+
 ########## TRANSMISSION #########
 
 msgctxt "#46000"
diff -Naurp Alexelec-settings-30c0064/language/Russian/strings.po Alexelec-settings-30c0064-patch/language/Russian/strings.po
--- Alexelec-settings-30c0064/language/Russian/strings.po	2018-02-01 18:55:11.000000000 +0200
+++ Alexelec-settings-30c0064-patch/language/Russian/strings.po	2018-02-01 19:09:07.000000000 +0200
@@ -1038,6 +1038,44 @@ msgctxt "#2851"
 msgid "Select the shell loaded at startup."
 msgstr "Выберите оболочку загружаемую при старте системы."
 
+########## TVIP #########
+
+msgctxt "#32860"
+msgid "TVIP configuration"
+msgstr "Настройки TVIP"
+
+msgctxt "#32861"
+msgid "Enable playlist"
+msgstr "Использовать плейлист"
+
+msgctxt "#2861"
+msgid "Use the specified channel source (m3u-playlist)."
+msgstr "Использовать определенный плейлист (файл *.m3u)."
+
+msgctxt "#32862"
+msgid "M3U file:"
+msgstr "M3U файл:"
+
+msgctxt "#2862"
+msgid "Select *.m3u file."
+msgstr "Выберите *.m3u файл."
+
+msgctxt "#32863"
+msgid "Update playlist"
+msgstr "Обновлять плейлист"
+
+msgctxt "#2863"
+msgid "Update Torrent-TV playlist before TVIP start."
+msgstr "Обновлять Торрент-ТВ плейлист перед стартом TVIP."
+
+msgctxt "#32864"
+msgid "First channel"
+msgstr "Канал #1"
+
+msgctxt "#2864"
+msgid "Start TVIP from the first channel."
+msgstr "Начинать показ с первого канала в списке."
+
 ########## ACESTREAM #########
 
 msgctxt "#34000"
@@ -1273,21 +1311,45 @@ msgid "Use external IP-address in playli
 msgstr "Использовать IP-адрес внешнего интерфейса при создании плейлиста."
 
 msgctxt "#34062"
-msgid "For"
-msgstr "Для"
+msgid "For:"
+msgstr "Для:"
 
 msgctxt "#3462"
 msgid "Playlist for AceStream and/or Noxbit."
 msgstr "Плейлист для AceStream и/или Noxbit."
 
 msgctxt "#34063"
-msgid "Execute..."
-msgstr "Выполнить..."
+msgid "Execute ->"
+msgstr "Выполнить ->"
 
 msgctxt "#3463"
 msgid "Create a playlist for Torrent-TV (/storage/ttv-m3u/)."
 msgstr "Создать плейлист для Торрент-ТВ (/storage/ttv-m3u/)."
 
+msgctxt "#34064"
+msgid "Login"
+msgstr "Логин"
+
+msgctxt "#3464"
+msgid "Login from site http://torrent-tv.ru"
+msgstr "Имя пользователя на сайте http://torrent-tv.ru"
+
+msgctxt "#34065"
+msgid "Password"
+msgstr "Пароль"
+
+msgctxt "#3465"
+msgid "Password from site http://torrent-tv.ru"
+msgstr "Пароль пользователя на сайте http://torrent-tv.ru"
+
+msgctxt "#34066"
+msgid "Channel Source:"
+msgstr "Источник каналов:"
+
+msgctxt "#3466"
+msgid "Channel source for AceStream: Torrent-TV, AceStream or both (All)."
+msgstr "Источник каналов для AceStream: Torrent-TV, AceStream или оба (All)."
+
 ########## TRANSMISSION #########
 
 msgctxt "#46000"
diff -Naurp Alexelec-settings-30c0064/src/defaults.py Alexelec-settings-30c0064-patch/src/defaults.py
--- Alexelec-settings-30c0064/src/defaults.py	2018-02-01 18:55:11.000000000 +0200
+++ Alexelec-settings-30c0064-patch/src/defaults.py	2018-02-01 19:00:52.000000000 +0200
@@ -1,8 +1,8 @@
+# -*- coding: utf-8 -*-
 ################################################################################
 #      This file is part of Alex@ELEC - http://www.alexelec.in.ua
-#      Copyright (C) 2011-2017 Alexandr Zuyev (alex@alexelec.in.ua)
+#      Copyright (C) 2011-present Alexandr Zuyev (alex@alexelec.in.ua)
 ################################################################################
-# -*- coding: utf-8 -*-
 
 import os
 import xbmcaddon
@@ -61,6 +61,11 @@ services = {
     'RUN_DISPCAP'       : "%s/dispcap.sh" % SCRIPT_DIR,
     # SHELL
     'D_SHELL'           : 'KODI',
+    # TVIP
+    'D_TVIP_ACT_PATCH'  : '0',
+    'D_TVIP_M3U'        : '',
+    'D_TVIP_UPDATE'     : '0',
+    'D_TVIP_LAST'       : '0',
     }
 
 ################################################################################
@@ -100,18 +105,22 @@ aceplist = {
     'D_CAT_FILMS'    : '1',
     'D_CAT_EROS'     : '1',
     'D_CAT_NEWS'     : '1',
-    'D_CAT_REGION'   : '1',
-    'D_CAT_MUSIC'    : '1',
-    'D_CAT_CHILDREN' : '1',
-    'D_CAT_SPORT'    : '1',
-    'D_CAT_RELIGION' : '1',
-    'D_CAT_MAN'      : '1',
-    'D_CAT_LEARN'    : '1',
+    'D_CAT_REGION'   : '0',
+    'D_CAT_MUSIC'    : '0',
+    'D_CAT_CHILDREN' : '0',
+    'D_CAT_SPORT'    : '0',
+    'D_CAT_RELIGION' : '0',
+    'D_CAT_MAN'      : '0',
+    'D_CAT_LEARN'    : '0',
     'D_CAT_ALLFON'   : '0',
     'D_ACETTV_IP'    : '0',
-    'D_TTV_SOFT'     : 'All',
+    'D_TTV_LOGIN'    : '',
+    'D_TTV_PASSW'    : '',
+    'D_TTV_SOFT'     : 'AceStream',
+    'D_TRANSLATE'    : 'Torrent-TV',
     'TTV_DEL_LIST'   : "rm -fr /storage/.config/acestream/ttv-m3u/*",
     'TTV_RUN_LIST'   : "/usr/bin/live-ttv-get",
+    'TTV_RUN_DIRECT' : "/usr/bin/ttv-direct-get",
     }
 
 ################################################################################
diff -Naurp Alexelec-settings-30c0064/src/resources/lib/modules/aceplist.py Alexelec-settings-30c0064-patch/src/resources/lib/modules/aceplist.py
--- Alexelec-settings-30c0064/src/resources/lib/modules/aceplist.py	2017-11-06 16:44:25.000000000 +0200
+++ Alexelec-settings-30c0064-patch/src/resources/lib/modules/aceplist.py	2018-02-01 15:14:12.000000000 +0200
@@ -1,8 +1,8 @@
+# -*- coding: utf-8 -*-
 ################################################################################
 #      This file is part of Alex@ELEC - http://www.alexelec.in.ua
-#      Copyright (C) 2011-2017 Alexandr Zuyev (alex@alexelec.in.ua)
+#      Copyright (C) 2011-present Alexandr Zuyev (alex@alexelec.in.ua)
 ################################################################################
-# -*- coding: utf-8 -*-
 
 import os
 import glob
@@ -27,9 +27,13 @@ class aceplist:
     D_CAT_LEARN = None
     D_CAT_ALLFON = None
     D_ACETTV_IP = None
+    D_TTV_LOGIN = None
+    D_TTV_PASSW = None
     D_TTV_SOFT = None
+    D_TRANSLATE = None
     TTV_DEL_LIST = None
-    TTV_RUN_LIST =None
+    TTV_RUN_LIST = None
+    TTV_RUN_DIRECT = None
 
     menu = {'99': {
         'name': 34050,
@@ -90,7 +94,7 @@ class aceplist:
                         'cat_region': {
                             'order': 6,
                             'name': 'Региональные',
-                            'value': '1',
+                            'value': '0',
                             'action': 'initialize_acettv',
                             'type': 'bool',
                             'InfoText': 3451,
@@ -98,7 +102,7 @@ class aceplist:
                         'cat_music': {
                             'order': 7,
                             'name': 'Музыка',
-                            'value': '1',
+                            'value': '0',
                             'action': 'initialize_acettv',
                             'type': 'bool',
                             'InfoText': 3451,
@@ -106,7 +110,7 @@ class aceplist:
                         'cat_children': {
                             'order': 8,
                             'name': 'Детские',
-                            'value': '1',
+                            'value': '0',
                             'action': 'initialize_acettv',
                             'type': 'bool',
                             'InfoText': 3451,
@@ -114,7 +118,7 @@ class aceplist:
                         'cat_sport': {
                             'order': 9,
                             'name': 'Спорт',
-                            'value': '1',
+                            'value': '0',
                             'action': 'initialize_acettv',
                             'type': 'bool',
                             'InfoText': 3451,
@@ -122,7 +126,7 @@ class aceplist:
                         'cat_religion': {
                             'order': 10,
                             'name': 'Религиозные',
-                            'value': '1',
+                            'value': '0',
                             'action': 'initialize_acettv',
                             'type': 'bool',
                             'InfoText': 3451,
@@ -130,7 +134,7 @@ class aceplist:
                         'cat_man': {
                             'order': 11,
                             'name': 'Мужские',
-                            'value': '1',
+                            'value': '0',
                             'action': 'initialize_acettv',
                             'type': 'bool',
                             'InfoText': 3451,
@@ -138,7 +142,7 @@ class aceplist:
                         'cat_learn': {
                             'order': 12,
                             'name': 'Познавательные',
-                            'value': '1',
+                            'value': '0',
                             'action': 'initialize_acettv',
                             'type': 'bool',
                             'InfoText': 3451,
@@ -166,17 +170,42 @@ class aceplist:
                             'type': 'bool',
                             'InfoText': 3461,
                             },
-                        'ttv_soft': {
+                       'ttv_login': {
                             'order': 2,
+                            'name': 34064,
+                            'value': None,
+                            'action': 'initialize_acerun',
+                            'type': 'text',
+                            'InfoText': 3464,
+                            },
+                        'ttv_passw': {
+                            'order': 3,
+                            'name': 34065,
+                            'value': None,
+                            'action': 'initialize_acerun',
+                            'type': 'text',
+                            'InfoText': 3465,
+                            },
+                        'ttv_soft': {
+                            'order': 4,
                             'name': 34062,
-                            'value': 'All',
-                            'values': ['All', 'AceStream', 'Noxbit'],
+                            'value': 'AceStream',
+                            'values': ['All', 'AceDirect', 'AceStream', 'Noxbit'],
                             'action': 'initialize_acerun',
                             'type': 'multivalue',
                             'InfoText': 3462,
                             },
+                        'translate': {
+                            'order': 5,
+                            'name': 34066,
+                            'value': 'Torrent-TV',
+                            'values': ['All', 'AceStream', 'Torrent-TV'],
+                            'action': 'initialize_acerun',
+                            'type': 'multivalue',
+                            'InfoText': 3466,
+                            },
                         'acettv_run': {
-                            'order': 3,
+                            'order': 6,
                             'name': 34063,
                             'value': '0',
                             'action': 'execute_acettv',
@@ -278,12 +307,21 @@ class aceplist:
             self.oe.get_service_option('acettv', 'CAT_ALLFON', self.D_CAT_ALLFON).replace('"', '')
 
             # TTV-LIST Create
+            self.struct['acerun']['settings']['ttv_login']['value'] = \
+            self.oe.get_service_option('acerun', 'TTV_LOGIN', self.D_TTV_LOGIN).replace('"', '')
+
+            self.struct['acerun']['settings']['ttv_passw']['value'] = \
+            self.oe.get_service_option('acerun', 'TTV_PASSW', self.D_TTV_PASSW).replace('"', '')
+
             self.struct['acerun']['settings']['ttv_soft']['value'] = \
             self.oe.get_service_option('acerun', 'TTV_SOFT', self.D_TTV_SOFT).replace('"', '')
 
             self.struct['acerun']['settings']['acettv_ip']['value'] = \
             self.oe.get_service_option('acerun', 'ACETTV_IP', self.D_ACETTV_IP).replace('"', '')
 
+            self.struct['acerun']['settings']['translate']['value'] = \
+            self.oe.get_service_option('acerun', 'TRANSLATE', self.D_TRANSLATE).replace('"', '')
+
             self.oe.dbg_log('aceplist::load_values', 'exit_function', 0)
         except Exception, e:
             self.oe.dbg_log('aceplist::load_values', 'ERROR: (%s)' % repr(e))
@@ -355,6 +393,9 @@ class aceplist:
             state = 1
             options['TTV_SOFT'] = '"%s"' % self.struct['acerun']['settings']['ttv_soft']['value']
             options['ACETTV_IP'] = '"%s"' % self.struct['acerun']['settings']['acettv_ip']['value']
+            options['TTV_LOGIN'] = '"%s"' % self.struct['acerun']['settings']['ttv_login']['value']
+            options['TTV_PASSW'] = '"%s"' % self.struct['acerun']['settings']['ttv_passw']['value']
+            options['TRANSLATE'] = '"%s"' % self.struct['acerun']['settings']['translate']['value']
             self.oe.set_service('acerun', options, state)
             self.oe.set_busy(0)
             self.oe.dbg_log('aceplist::initialize_acerun', 'exit_function', 0)
@@ -365,19 +406,36 @@ class aceplist:
     def execute_acettv(self, listItem=None):
         try:
             self.oe.dbg_log('aceplist::execute_acettv', 'enter_function', 0)
-            if os.path.exists(self.TTV_RUN_LIST):
-                self.oe.notify(self.oe._(32363), 'Create TTV playlist...')
-                self.oe.set_busy(1)
-                message = self.oe.execute(self.TTV_RUN_LIST, 1).strip()
-                self.oe.set_busy(0)
-                if message == 'Done! Playlist created.':
-                    dialog = xbmcgui.Dialog()
-                    ret = dialog.yesno('TTV playlist', 'Playlist created. Restart Kodi?')
-                    if ret:
-                        xbmc.executebuiltin('RestartApp')
-                else:
-                    dialog = xbmcgui.Dialog()
-                    dialog.notification('TTV playlist',
+            if self.struct['acerun']['settings']['ttv_soft']['value'] != 'AceDirect':
+                if os.path.exists(self.TTV_RUN_LIST):
+                    self.oe.notify(self.oe._(32363), 'Create TTV playlist...')
+                    self.oe.set_busy(1)
+                    message = self.oe.execute(self.TTV_RUN_LIST, 1).strip()
+                    self.oe.set_busy(0)
+                    if message == 'Done! Playlist created.':
+                        dialog = xbmcgui.Dialog()
+                        ret = dialog.yesno('TTV playlist', 'Playlist created. Restart Kodi?')
+                        if ret:
+                            xbmc.executebuiltin('RestartApp')
+                    else:
+                        dialog = xbmcgui.Dialog()
+                        dialog.notification('TTV playlist',
+                                        '%s' % message,
+                                        xbmcgui.NOTIFICATION_INFO, 3000)
+            else:
+                if os.path.exists(self.TTV_RUN_DIRECT):
+                    self.oe.notify(self.oe._(32363), 'Create AceStream playlist torrent-tv...')
+                    self.oe.set_busy(1)
+                    message = self.oe.execute(self.TTV_RUN_DIRECT, 1).strip()
+                    self.oe.set_busy(0)
+                    if message == 'Done! Playlist created.':
+                        dialog = xbmcgui.Dialog()
+                        ret = dialog.yesno('AceStream Torrent-TV playlist', 'Playlist created. Restart Kodi?')
+                        if ret:
+                            xbmc.executebuiltin('RestartApp')
+                    else:
+                        dialog = xbmcgui.Dialog()
+                        dialog.notification('AceStream Torrent-TV playlist',
                                         '%s' % message,
                                         xbmcgui.NOTIFICATION_INFO, 3000)
 
diff -Naurp Alexelec-settings-30c0064/src/resources/lib/modules/services.py Alexelec-settings-30c0064-patch/src/resources/lib/modules/services.py
--- Alexelec-settings-30c0064/src/resources/lib/modules/services.py	2018-02-01 18:55:11.000000000 +0200
+++ Alexelec-settings-30c0064-patch/src/resources/lib/modules/services.py	2018-02-01 17:05:17.000000000 +0200
@@ -1,8 +1,8 @@
+# -*- coding: utf-8 -*-
 ################################################################################
 #      This file is part of Alex@ELEC - http://www.alexelec.in.ua
-#      Copyright (C) 2011-2017 Alexandr Zuyev (alex@alexelec.in.ua)
+#      Copyright (C) 2011-present Alexandr Zuyev (alex@alexelec.in.ua)
 ################################################################################
-# -*- coding: utf-8 -*-
 
 import os
 import glob
@@ -45,6 +45,10 @@ class services:
     RUN_DISPCAP = None
 
     D_SHELL = None
+    D_TVIP_M3U = None
+    D_TVIP_LAST = None
+    D_TVIP_ACT_PATCH = None
+    D_TVIP_UPDATE = None
 
     menu = {'1': {
         'name': 32001,
@@ -437,6 +441,49 @@ class services:
                             },
                         },
                     },
+                'tvip': {
+                    'order': 15,
+                    'name': 32860,
+                    'not_supported': [],
+                    'settings': {
+                        'tvip_act_patch': {
+                            'order': 1,
+                            'name': 32861,
+                            'value': '0',
+                            'action': 'initialize_tvip',
+                            'type': 'bool',
+                            'InfoText': 2861,
+                            },
+                        'tvip_m3u': {
+                            'order': 2,
+                            'name': 32862,
+                            'value': '',
+                            'action': 'initialize_tvip',
+                            'type': 'file',
+                            'parent': {
+                                'entry': 'tvip_act_patch',
+                                'value': ['1']
+                                },
+                            'InfoText': 2862,
+                            },
+                        'tvip_update': {
+                            'order': 3,
+                            'name': 32863,
+                            'value': '0',
+                            'action': 'initialize_tvip',
+                            'type': 'bool',
+                            'InfoText': 2863,
+                            },
+                        'tvip_last': {
+                            'order': 4,
+                            'name': 32864,
+                            'value': '0',
+                            'action': 'initialize_tvip',
+                            'type': 'bool',
+                            'InfoText': 2864,
+                            },
+                        },
+                    },
                 }
 
             self.oe = oeMain
@@ -460,6 +507,7 @@ class services:
             self.initialize_nand()
             self.initialize_dispcap()
             self.initialize_shell()
+            self.initialize_tvip()
             self.oe.dbg_log('services::start_service', 'exit_function', 0)
         except Exception, e:
             self.oe.dbg_log('services::start_service', 'ERROR: (%s)' % repr(e))
@@ -589,6 +637,19 @@ class services:
             self.struct['shell']['settings']['first_shell']['value'] = \
             self.oe.get_service_option('shell', 'SHELL', self.D_SHELL).replace('"', '')
 
+            # TVIP
+            self.struct['tvip']['settings']['tvip_act_patch']['value'] = \
+            self.oe.get_service_option('tvip', 'TVIP_ACT_PATCH', self.D_TVIP_ACT_PATCH).replace('"', '')
+
+            self.struct['tvip']['settings']['tvip_m3u']['value'] = \
+            self.oe.get_service_option('tvip', 'TVIP_M3U', self.D_TVIP_M3U).replace('"', '')
+
+            self.struct['tvip']['settings']['tvip_update']['value'] = \
+            self.oe.get_service_option('tvip', 'TVIP_UPDATE', self.D_TVIP_UPDATE).replace('"', '')
+
+            self.struct['tvip']['settings']['tvip_last']['value'] = \
+            self.oe.get_service_option('tvip', 'TVIP_LAST', self.D_TVIP_LAST).replace('"', '')
+
             self.oe.dbg_log('services::load_values', 'exit_function', 0)
         except Exception, e:
             self.oe.dbg_log('services::load_values', 'ERROR: (%s)' % repr(e))
@@ -915,6 +976,25 @@ class services:
             self.oe.set_busy(0)
             self.oe.dbg_log('services::initialize_shell', 'ERROR: (%s)' % repr(e), 4)
 
+    def initialize_tvip(self, **kwargs):
+        try:
+            self.oe.dbg_log('services::initialize_tvip', 'enter_function', 0)
+            self.oe.set_busy(1)
+            if 'listItem' in kwargs:
+                self.set_value(kwargs['listItem'])
+            options = {}
+            state = 1
+            options['TVIP_ACT_PATCH'] = '"%s"' % self.struct['tvip']['settings']['tvip_act_patch']['value']
+            options['TVIP_M3U'] = '"%s"' % self.struct['tvip']['settings']['tvip_m3u']['value']
+            options['TVIP_UPDATE'] = '"%s"' % self.struct['tvip']['settings']['tvip_update']['value']
+            options['TVIP_LAST'] = '"%s"' % self.struct['tvip']['settings']['tvip_last']['value']
+            self.oe.set_service('tvip', options, state)   
+            self.oe.set_busy(0)
+            self.oe.dbg_log('services::initialize_tvip', 'exit_function', 0)
+        except Exception, e:
+            self.oe.set_busy(0)
+            self.oe.dbg_log('services::initialize_tvip', 'ERROR: (%s)' % repr(e), 4)
+
     def execute_dispcap(self, listItem=None):
         try:
             self.oe.dbg_log('services::execute_dispcap', 'enter_function', 0)
