diff -Naurp retroarch-ce9439e/frontend/drivers/platform_linux.c retroarch-ce9439e-patch/frontend/drivers/platform_linux.c
--- retroarch-ce9439e/frontend/drivers/platform_linux.c	2017-05-13 14:12:15.000000000 +0300
+++ retroarch-ce9439e-patch/frontend/drivers/platform_linux.c	2017-05-13 15:22:42.000000000 +0300
@@ -1205,8 +1205,7 @@ static void frontend_linux_get_os(char *
 #endif
 }
 
-#ifdef HAVE_LAKKA
-static void frontend_linux_get_lakka_version(char *s,
+static void frontend_linux_get_ae_version(char *s,
       size_t len)
 {
    char version[128];
@@ -1223,7 +1222,6 @@ static void frontend_linux_get_lakka_ver
 
    pclose(command_file);
 }
-#endif
 
 static void frontend_linux_get_env(int *argc,
       char *argv[], void *data, void *params_data)
@@ -2151,9 +2149,7 @@ frontend_ctx_driver_t frontend_ctx_linux
    frontend_linux_destroy_signal_handler_state,
    NULL,                         /* attach_console */
    NULL,                         /* detach_console */
-#ifdef HAVE_LAKKA
-   frontend_linux_get_lakka_version,    /* get_lakka_version */
-#endif
+   frontend_linux_get_ae_version,    /* get_ae_version */
 #ifdef ANDROID
    "android"
 #else
diff -Naurp retroarch-ce9439e/frontend/drivers/platform_null.c retroarch-ce9439e-patch/frontend/drivers/platform_null.c
--- retroarch-ce9439e/frontend/drivers/platform_null.c	2017-05-13 14:12:15.000000000 +0300
+++ retroarch-ce9439e-patch/frontend/drivers/platform_null.c	2017-05-13 15:23:04.000000000 +0300
@@ -41,8 +41,6 @@ frontend_ctx_driver_t frontend_ctx_null
    NULL,                         /* destroy_sighandler_state */
    NULL,                         /* attach_console */
    NULL,                         /* detach_console */
-#ifdef HAVE_LAKKA
-   NULL,                         /* get_lakka_version */
-#endif
+   NULL,                         /* get_ae_version */
    "null",
 };
diff -Naurp retroarch-ce9439e/frontend/frontend_driver.h retroarch-ce9439e-patch/frontend/frontend_driver.h
--- retroarch-ce9439e/frontend/frontend_driver.h	2017-05-13 14:12:15.000000000 +0300
+++ retroarch-ce9439e-patch/frontend/frontend_driver.h	2017-05-13 15:24:20.000000000 +0300
@@ -85,9 +85,7 @@ typedef struct frontend_ctx_driver
    void (*destroy_signal_handler_state)(void);
    void (*attach_console)(void);
    void (*detach_console)(void);
-#ifdef HAVE_LAKKA
-   void (*get_lakka_version)(char *, size_t);
-#endif
+   void (*get_ae_version)(char *, size_t);
 
    const char *ident;
 
diff -Naurp retroarch-ce9439e/intl/msg_hash_us.h retroarch-ce9439e-patch/intl/msg_hash_us.h
--- retroarch-ce9439e/intl/msg_hash_us.h	2017-05-13 14:12:15.000000000 +0300
+++ retroarch-ce9439e-patch/intl/msg_hash_us.h	2017-05-13 15:26:07.000000000 +0300
@@ -1391,8 +1391,8 @@ MSG_HASH(MENU_ENUM_LABEL_VALUE_SYSTEM_IN
       "JACK support")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_SYSTEM_INFO_KMS_SUPPORT,
       "KMS/EGL support")
-MSG_HASH(MENU_ENUM_LABEL_VALUE_SYSTEM_INFO_LAKKA_VERSION,
-      "Lakka Version")
+MSG_HASH(MENU_ENUM_LABEL_VALUE_SYSTEM_INFO_AE_VERSION,
+      "Alex@ELEC Version")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_SYSTEM_INFO_LIBRETRODB_SUPPORT,
       "LibretroDB support")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_SYSTEM_INFO_LIBUSB_SUPPORT,
diff -Naurp retroarch-ce9439e/menu/menu_displaylist.c retroarch-ce9439e-patch/menu/menu_displaylist.c
--- retroarch-ce9439e/menu/menu_displaylist.c	2017-05-13 14:12:15.000000000 +0300
+++ retroarch-ce9439e-patch/menu/menu_displaylist.c	2017-05-13 15:28:57.000000000 +0300
@@ -679,23 +679,21 @@ static int menu_displaylist_parse_system
             MENU_ENUM_LABEL_SYSTEM_INFO_ENTRY,
             MENU_SETTINGS_CORE_INFO_NONE, 0, 0);
 
-#ifdef HAVE_LAKKA
-      if (frontend->get_lakka_version)
+      if (frontend->get_ae_version)
       {
-         frontend->get_lakka_version(tmp2, sizeof(tmp2));
+         frontend->get_ae_version(tmp2, sizeof(tmp2));
 
          fill_pathname_noext(tmp,
-               msg_hash_to_str(MENU_ENUM_LABEL_VALUE_SYSTEM_INFO_LAKKA_VERSION),
+               msg_hash_to_str(MENU_ENUM_LABEL_VALUE_SYSTEM_INFO_AE_VERSION),
                ": ",
                sizeof(tmp));
-         strlcat(tmp, frontend->get_lakka_version ?
+         strlcat(tmp, frontend->get_ae_version ?
                tmp2 : msg_hash_to_str(MENU_ENUM_LABEL_VALUE_NOT_AVAILABLE),
                sizeof(tmp));
          menu_entries_append_enum(info->list, tmp, "",
                MENU_ENUM_LABEL_SYSTEM_INFO_ENTRY,
                MENU_SETTINGS_CORE_INFO_NONE, 0, 0);
       }
-#endif
 
       if (frontend->get_name)
       {
diff -Naurp retroarch-ce9439e/msg_hash.h retroarch-ce9439e-patch/msg_hash.h
--- retroarch-ce9439e/msg_hash.h	2017-05-13 14:12:15.000000000 +0300
+++ retroarch-ce9439e-patch/msg_hash.h	2017-05-13 15:30:01.000000000 +0300
@@ -1382,7 +1382,7 @@ enum msg_hash_enums
 
    /* System information */
 
-   MENU_ENUM_LABEL_VALUE_SYSTEM_INFO_LAKKA_VERSION,
+   MENU_ENUM_LABEL_VALUE_SYSTEM_INFO_AE_VERSION,
    MENU_ENUM_LABEL_VALUE_SYSTEM_INFO_DYNAMIC_SUPPORT,
    MENU_ENUM_LABEL_VALUE_SYSTEM_INFO_BUILD_DATE,
    MENU_ENUM_LABEL_VALUE_SYSTEM_INFO_GIT_VERSION,
